hawkular-inventory-impl-tinkerpop-parent/hawkular-inventory-impl-tinkerpop/src/main/java/org/hawkular/inventory/impl/tinkerpop/TinkerpopBackend.java

```diff
@@ -178,7 +179,7 @@ public Element find(CanonicalPath element) throws ElementNotFoundException {
                     });
         }
 
-        return new Page<>(q2.toList(), pager, q.getCount("total"));
+        return new Page<>(q2, pager, q.getCount("total"));
```

Comment:
Btw this starts to smell like we'll be porting the backend to rx, like metrics did. That would make these kind of API problems more explicit from the user POV and therefore the users would use the API differently than what we are trying to offer now.